File: LeetCode/370-range-addition/ex_370.py
class Solution(object):
    def getModifiedArray(self, length, updates):
        """
        :type length: int
        :type updates: List[List[int]]
        :rtype: List[int]
        """
        res = [0] * length
        for start, end, val in updates:
            res[start] += val
            if end + 1 < length:
                res[end + 1] -= val
        for i in xrange(1, length):
            res[i] += res[i-1]
        return res

File: LeetCode/370-range-addition/ex_370.cpp
class Solution {
public:
    vector<int> getModifiedArray(int length, vector<vector<int>>& updates) {
        vector<int> res(length, 0);
        for (auto &p : updates) {
            int start = p[0], end = p[1], val = p[2];
            res[start] += val;
            if (end + 1 < length)
                res[end + 1] -= val;
        }
        // for (int i = 1; i != length; ++i)
        //     res[i] += res[i - 1];
        partial_sum(res.begin(), res.end(), res.begin());
        return res;
    }
};